### ISSUE TITLE
Wallaby.js + Flow.js Inline Comments = "Error: Failed to instrument"

### ISSUE THREAD

**Issue by ancms2600** [wallabyjs/public#2481]
### Issue description or question

When using [Facebook Flow Comment Types](https://flow.org/en/docs/types/comments/), basically you are using comment blocks `/**/` everywhere. Wallaby is fine with that everywhere except if you place a comment between the `function name` and its `(arg, list) {`. This is used for [Generic Types](https://flow.org/en/docs/types/generics/). See example below.

The odd thing is that [acorn seems fine with it](https://astexplorer.net/), so this must be an issue introduced by Wallaby after AST parsing?

```js
// @flow strict

class A {
    B/*c*/() {
    }
};
```
produces error:

```
Error: Failed to instrument example.js
  3 |   B
  4 |   /*c*/
> 5 |   () {}
    |   ^ SyntaxError: Unexpected token (5:2)
  6 | 
  7 | }
  8 | 
```

### Wallaby diagnostics report

```js
{
  editorVersion: '1.48.0-insider',
  pluginVersion: '1.0.217',
  editorType: 'VSCode',
  osVersion: 'linux 5.5.0-kali1-amd64',
  nodeVersion: 'v12.18.2',
  coreVersion: '1.0.926',
  packageJSON: {
    devDependencies: { 'flow-bin': '~0.128.0', jest: '~26' }
  }
}
```

**Comment by ancms2600** [comment]
Separate issue, but while I'm at it, I noticed acorn doesn't support class fields. Lame. Had to remove those to use Wallaby :(

```js
class A {
  // none of these allowed
  b;
  static c;
  d = 1;
  static e = 2;
}
```

I didn't check but private fields and methods are probably also not supported (not yet supported by Flow, either)

**Comment by ArtemGovorov** [comment]
Acorn (with plugins shipped with Wallaby) supports both scenarios. 
On a simple project without Babel with the latest version of node (supporting class fields):

<img width="260" alt="Screen Shot 2020-07-18 at 11 21 58 am" src="https://user-images.githubusercontent.com/979966/87841504-14325700-c8e9-11ea-872d-0b5dab9b90d6.png">

From the error:

```
Error: Failed to instrument example.js
  3 |   B
  4 |   /*c*/
> 5 |   () {}
    |   ^ SyntaxError: Unexpected token (5:2)
  6 | 
  7 | }
  8 | 
```

it looks like Babel (prior to Wallaby acorn run) is introducing a line break between `B` and the `/*c*/` comment, making the code invalid.

Unfortunately there's not enough information in your diagnostics report to reproduce the issue. Can you please:
- check that running `npx jest` from your repo's root works and doesn't produce the same errors,
- share your Jest config, your Babel config and your Wallaby config (unless you are using automatic configuration).

**Comment by ArtemGovorov** [comment]
Please disregard my previous comment regarding the comment scenario:

```
class A {
    B/*c*/() {
    }
};
```

While Babel indeed is inserting the line break, it should not be breaking acorn. We will be investigating why it happens.

Regarding 
> Separate issue, but while I'm at it, I noticed acorn doesn't support class fields. Lame. Had to remove those to use Wallaby :(

Wallaby acorn support it, so maybe you are missing Babel plugin `@babel/plugin-proposal-class-properties` to let babel transpile them? If not, what errors are you getting from Wallaby when you are trying to use class fields?

**Comment by ArtemGovorov** [comment]
We have fixed this issue:

```
// @flow strict

class A {
    B/*c*/() {
    }
};
```

```
Error: Failed to instrument example.js
  3 |   B
  4 |   /*c*/
> 5 |   () {}
    |   ^ SyntaxError: Unexpected token (5:2)
  6 | 
  7 | }
  8 | 
```

in [the latest core 1.0.927](https://wallabyjs.com/docs/intro/troubleshooting.html#forcing-wallabyjs-core-update).

The other issue you've raised (class fields) doesn't seem to be related to Wallaby (Wallaby ships with acorn plugin that supports class fields). If you can share the error source/message that you are getting, we are happy to investigate.

**Comment by ancms2600** [comment]
there is a second case:

```js
class A {
    static B/*c*/() { // same error as before
    }
};
```

confirmed fixed for instance methods, though.

```js
class A {
    B/*c*/() {  // works now
    }
};
```

**Comment by ArtemGovorov** [comment]
Thanks for checking. The 

```js
class A {
    static B/*c*/() { // same error as before
    }
};
```

is now fixed and the fix is published in [the latest core v1.0.928](https://wallabyjs.com/docs/intro/troubleshooting.html#forcing-wallabyjs-core-update).

**Comment by ancms2600** [comment]
Confirmed working. Also, the class fields are working! 

Thanks. I am a happy camper now! :)
